### PR TITLE
Add `enableTracing` option

### LINKED ISSUE #2522: Add enableTracing option
A new simplified options for configuring to turn tracing & performance on should be added to all SDKs. For each SDK the default sample rate should be agreed upon.  

Docs do not need to be updated yet, nor the onboarding wizard. This can wait until the top level options are added to a number of SDKs and we decide if we want to configure the default onboarding experience, for all new users.

See [develop docs](https://develop.sentry.dev/sdk/performance/#enabletracing) for more information. This is a non-breaking change to be clear. Any of the three options should enable performance, alone or in combination

### Open point:
recommend default it 1.0 or 100%. Can discuss if it should be less.

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

Adds a new `enableTracing` option. If set to true, performance is enabled, even if no `tracesSampleRate` or `tracesSampler` have been configured. For now we set a default `tracesSampleRate` of `1.0` if none has been configured explicitly. If `enableTracing` is set to `false` performance is disabled, regardless of `tracesSampleRate` and `tracesSampler` options. The default for `enableTracing` is `null` meaning existing behaviour remains unchanged (setting either `tracesSampleRate` or `tracesSampler` enables performance)

## :bulb: Motivation and Context

Closes #2522

## :green_heart: How did you test it?
Unit tests, manually using Android and Spring Boot (Jakarta)

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed. - Will be done for all SDKs at once.
- [-] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [?] No breaking change for hybrid SDKs or communicated to hybrid SDKs.
	- 	Shouldn't break anything, existing behaviour should still work

### CODE REVIEW

**Comment by adinauer** [comment]
Here's the test for PHP documenting how the options combine: https://github.com/getsentry/sentry-php/pull/1458/files#diff-fa1abff32fd8afc425213dea903ea43386633e2123cc0611c7c2582811d8cffeR631-R653

File: sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java [hunk 1]
```diff
@@ -333,6 +338,23 @@ private static boolean readBool(
     return value;
   }
 
+  @SuppressWarnings("deprecation")
+  private static @Nullable Boolean readBoolNullable(
```
Comment (adinauer) [inline sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java:342]: Alternative could be to read `metadata.getBoolean()` twice with different defaults and check if the result differs, then return `null`. Lmk if you prefer that or have a better suggestion.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ### Features
 
 - Add `main` flag to threads and `in_foreground` flag for app contexts  ([#2516](https://github.com/getsentry/sentry-java/pull/2516))
+- Add `enableTracing` option ([#2530](https://github.com/getsentry/sentry-java/pull/2530))
```
Comment (romtsn) [inline CHANGELOG.md:8]: `m`: can we elaborate a little bit more for the changelog readers how it behaves? (i.e. it does not change existing default behavior, etc.)

File: sentry/src/main/java/io/sentry/SentryOptions.java [hunk 3]
```diff
@@ -818,6 +821,24 @@ public void setSampleRate(Double sampleRate) {
...
+  /**
+   * Whether generation of transactions and propagation of trace data is enabled.
+   *
+   * <p>NOTE: There is also {@link SentryOptions#isTracingEnabled()} which checks other options as
+   * well.
+   *
+   * @return true if enabled, false if disabled, null can mean enabled if {@link
+   *     SentryOptions#getTracesSampleRate()} or {@link SentryOptions#getTracesSampler()} are set.
+   */
+  public @Nullable Boolean isEnableTracing() {
```
[the remark this answers is not in the document]
Comment (romtsn) [reply sentry/src/main/java/io/sentry/SentryOptions.java:833]: maybe we should change it to `getEnableTracing()`? Does not align with the rest of `isSomething()` options, but I think it's reasonable to break the naming convention here in favor of clarity

File: sentry/src/test/java/io/sentry/SentryOptionsTest.kt [hunk 4]
```diff
@@ -114,6 +114,51 @@ class SentryOptionsTest {
         assertFailsWith<IllegalArgumentException> { SentryOptions().tracesSampleRate = -0.0000000000001 }
     }
 
+    @Test
```
Comment (romtsn) [inline sentry/src/test/java/io/sentry/SentryOptionsTest.kt:117]: `m`: what if we reuse the test data from the PHP sdk PR and create a new Parameterized class test, so it's 100% aligned?

**Review by romtsn — APPROVED** [review]
a couple of small things, but LGTM otherwise, thanks 🚀

File: sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java (hunk 1 above)
Comment (adinauer) [reply sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java:342]: You mean keep as is or the read twice variant?

File: sentry/src/main/java/io/sentry/SentryOptions.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (adinauer) [reply sentry/src/main/java/io/sentry/SentryOptions.java:833]: Will change it.

File: CHANGELOG.md (hunk 2 above)
Comment (adinauer) [reply CHANGELOG.md:8]: Will do

File: sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java (hunk 1 above)
Comment (romtsn) [reply sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java:342]: keep as is

File: sentry/src/test/java/io/sentry/SentryOptionsTest.kt (hunk 4 above)
Comment (adinauer) [reply sentry/src/test/java/io/sentry/SentryOptionsTest.kt:117]: Added a new test called `SentryOptionsTracingTest`

File: CHANGELOG.md (hunk 2 above)
Comment (adinauer) [reply CHANGELOG.md:8]: updated

**Pull request merged**